pkg/ottl/functions.go

```diff
@@ -17,15 +17,17 @@ package ottl // import "github.com/open-telemetry/opentelemetry-collector-contri
 import (
 	"fmt"
 	"reflect"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/internal/ottlgrammar"
 )
 
-type PathExpressionParser func(*Path) (GetSetter, error)
+type PathExpressionParser func(*ottlgrammar.Path) (GetSetter, error)
 
-type EnumParser func(*EnumSymbol) (*Enum, error)
+type EnumParser func(*ottlgrammar.EnumSymbol) (*Enum, error)
```

Comment:
So the intent is to not allow external packages to implement these funcs?

Reason I am asking, is because the argument of the func is in internal, so not a good design to have to interact with an internal type.

Reply:
Ya the goal is to hide away all the grammar structs and make them only part of parsing and nothing else.  Looks like I need to do some more refactoring to get them out of Contexts and ottl's exposed funcs/types.

Reply:
The other alternative is to leave them in ottl and not expose the type itself and only expose its fields, which is a requirement for the parsing library we use.

Reply:
I think the simplest solution would be to keep the grammar types in the `ottl` package, but only export types that are user-facing. It's likely that the grammar should stay small enough that a single `grammar.go` file should suffice. Additionally, I think it would simplify things from the user's perspective if all the types necessary to instantiate a parser are available from a single package.